File: Programmers/[12900] 2 x n 타일링/ONE/12900.java
class Solution {
    public int solution(int n) {
        // 피보나치 수열의 규칙
        int[] fibonacci = new int[n + 1];
        fibonacci[0] = 1;
        fibonacci[1] = 1;

        for (int i = 2; i <= n; i++)
            fibonacci[i] = (fibonacci[i - 2] + fibonacci[i - 1]) % 1000000007;

        return fibonacci[n];
    }
}

File: Programmers/[12900] 2 x n 타일링/woong-jae/12900.js
function solution(n) {
  const dp = Array(n + 1);
  dp[1] = 1;
  dp[2] = 2;
  
  for(let i = 3; i <= n; i++) {
      dp[i] = (dp[i - 1] + dp[i - 2]) % 1000000007;
  }
  
  return dp[n];
}